Angular/src/app/sidebar/sidebar.component.ts

```diff
@@ -1,15 +1,24 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit } from "@angular/core";
+import { TranslateService } from '@ngx-translate/core';
 
 @Component({
-  selector: 'app-sidebar',
-  templateUrl: './sidebar.component.html',
-  styleUrls: ['./sidebar.component.scss']
+  selector: "app-sidebar",
+  templateUrl: "./sidebar.component.html",
+  styleUrls: ["./sidebar.component.scss"]
 })
+
 export class SidebarComponent implements OnInit {
+  constructor(public translate: TranslateService) {}
```

Comment:
Why does this constructor take a TranslateService? I'm just not familiar with what this is doing

Reply:
@jakobh7 Because this component needs to use this service for changing the locale. It's the way angular do a dependency injection.